src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindSmallIndelRegions.java

```diff
@@ -0,0 +1,135 @@
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.barclay.argparser.ArgumentCollection;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.programgroups.StructuralVariationSparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.tools.spark.utils.IntHistogram;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.*;
+
+import static org.broadinstitute.hellbender.tools.spark.sv.StructuralVariationDiscoveryArgumentCollection.FindBreakpointEvidenceSparkArgumentCollection;
+
+@CommandLineProgramProperties(summary="Find regions likely to contain small indels due to fragment length anomalies.",
+        oneLineSummary="Find regions containing small indels.",
+        programGroup=StructuralVariationSparkProgramGroup.class)
+public final class FindSmallIndelRegions extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+
+    @ArgumentCollection
+    private static final FindBreakpointEvidenceSparkArgumentCollection params =
+                new FindBreakpointEvidenceSparkArgumentCollection();
+
+    private static final int EVIDENCE_SIZE_GUESS = 1000;
+
+    @Override
+    public boolean requiresReads()
+    {
+        return true;
+    }
+
+    @Override
+    protected void runTool( final JavaSparkContext ctx ) {
+
+        final JavaRDD<GATKRead> allReads = getUnfilteredReads();
+        final Set<Integer> crossContigIgnoreSet = Collections.emptySet();
+        final SAMFileHeader header = getHeaderForReads();
+        final int maxTrackedFragmentLength = params.maxTrackedFragmentLength;
+        final SVReadFilter filter = new SVReadFilter(params);
+        final ReadMetadata readMetadata =
+                new ReadMetadata(crossContigIgnoreSet, header, maxTrackedFragmentLength, allReads, filter);
+        final Broadcast<ReadMetadata> broadcastReadMetadata = ctx.broadcast(readMetadata);
+
+        final List<BreakpointEvidence> evidenceList =
+            allReads
+                .mapPartitions(readItr -> {
+                    final List<BreakpointEvidence> evList = new ArrayList<>(EVIDENCE_SIZE_GUESS);
+                    final Finder finder = new Finder(broadcastReadMetadata.value(), filter);
+                    while ( readItr.hasNext() ) {
+                        finder.test(readItr.next(), evList);
+                    }
+                    finder.checkHistograms(evList);
+                    return evList.iterator();
+                })
+                .collect();
+
+        for ( final BreakpointEvidence evidence : evidenceList ) {
+            System.out.println(evidence);
+        }
+    }
+
+    public static final class Finder {
+        private static final int BLOCK_SIZE = 250;
+        private static final int EVIDENCE_WEIGHT = 5;
+        private static final float KS_SIGNIFICANCE = 1.E-5f;
```

Comment:
What was your justification for setting this value? Do you have a plot of what they looked like empirically? We are doing a lot of tests across the genome.

It would likely be tricky or inefficient to implement, but if we could gather the p-values for many of these tests we could use a q-value FDR approach.